lib/traject/psulib_marc.rb

```diff
@@ -40,3 +40,19 @@ def process_subject_topic_facet(record, fields)
   end
   subjects.flatten
 end
+
+# for genre facet and display
+# limit to subfield $2 vocabularies for 655|*7 genres
+def process_genre(record, fields)
+  genres = []
+  vocabulary = %w[lcgft fast]
+  Traject::MarcExtractor.cached(fields).collect_matching_lines(record) do |field, spec, extractor|
+    genre = extractor.collect_subfields(field, spec).first
+    include_genre = true
+    unless genre.nil?
+      include_genre = vocabulary.include?(field['2'].to_s.downcase) if (field.tag == '655') && (field.indicator2 == '7')
```

Comment:
I guess it's a little weird to me that there's a check for `field.tag == '655'`. Like, the whole method is predicated on the logic built into the 655. Maybe it makes sense to omit it, or add a check at the beginning and bail then rather then proceed.

Reply:
Never mind, I think I see what's going on here, if it's a `650` you still want to add it, that's where `include_genre` will equal true.

Reply:
yeah, 650s should be included. I actually played around with the logic here to find a better way to implement and was waiting for your thoughts on it :)

Reply:
let me know if you come up with a better idea